### PR TITLE
Convert pasted text into markdown

### PR DESCRIPTION
**Pull request by tijmenb**
This is an experimental feature that might make it into the govspeak editor for content publisher. By trying it out here we can get some feedback from content designers who use this application.

This is a modified version of https://github.com/euangoddard/clipboard2markdown, updated to suit allow users to paste into a textarea directly.

I'm working from this reference file in Google Docs:

https://docs.google.com/document/d/1pNc4ScQ-w_jx_zccM0um5QHOFZua98bMx0r-sj3uC04/edit

https://trello.com/c/yKPf7Kd6

### CODE REVIEW

File: app/assets/javascripts/to-markdown.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      e.clipboardData.setData("text/html", markdown);
+      e.preventDefault();
+      output.insertAtCaret(markdown);
+      output.focus();
+    });
+  });
+})();
+
+// https://stackoverflow.com/questions/11076975/insert-text-into-textarea-at-cursor-position-javascript
+HTMLTextAreaElement.prototype.insertAtCaret = function (text) {
```
Comment (tijmenb) [inline app/assets/javascripts/to-markdown.js]: This doesn't seem to work on Firefox.

File: app/assets/javascripts/to-markdown.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+(function () {
+  'use strict';
+
+  // http://pandoc.org/README.html#pandocs-markdown
+  var pandoc = [
+    {
+      filter: 'h1',
+      replacement: function (content, node) {
+        var underline = Array(content.length + 1).join('=');
+        return '\n\n' + content + '\n' + underline + '\n\n';
```
Comment (tijmenb) [inline app/assets/javascripts/to-markdown.js]: `h1`s should be skipped.

File: app/assets/javascripts/to-markdown.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      }
+    },
+
+    {
+      filter: 'h2',
+      replacement: function (content, node) {
+        var underline = Array(content.length + 1).join('-');
+        return '\n\n' + content + '\n' + underline + '\n\n';
```
Comment (tijmenb) [inline app/assets/javascripts/to-markdown.js]: This should be `## {contents}`

**Comment by keithiopia** [comment]
Possible bug: I can't seem to copy from the box and paste into the same box.

**Comment by tijmenb** [comment]
Tracked here: https://trello.com/c/JzbiiIwn/154-allow-pasting-of-rich-text-into-the-govspeak-editor

**Pull request closed without merging**